src/rpc/methods/eth.rs

```diff
@@ -3169,6 +3174,116 @@ impl RpcMethod<2> for EthTraceReplayBlockTransactions {
     }
 }
 
+fn get_eth_block_number_from_string<DB: Blockstore>(
```

Comment:
We are already doing similar handling in https://github.com/ChainSafe/forest/blob/6708366b4b963793b0a8028780733e8c50d8e817/src/rpc/methods/eth.rs#L339-L348

Can the logic be unified?